src/qutip_qip/decompositions/single_decompositions.py

```diff
@@ -0,0 +1,226 @@
+import numpy as np
+import cmath
+
+from qutip.qobj import Qobj
+from qutip_qip.decompositions.general_decompositions import (check_input, check_input_shape,
+convert_qobj_gate_to_array, extract_global_phase, MethodError, GateError)
+
+from qutip_qip.operations import *
+from qutip_qip.circuit import QubitCircuit, Gate
+
+
+# Functions for decompose_to_rotation_matrices
+single_qubit = 1
```

Comment:
Should this line be here? Not sure what it is doing. :)

Reply:
This was added to differentiate the inputs to a decomposition function when the target and number of qubits was the same. I think, I can try to define this in every functions definition with a default value (like Boxi discusses[ here](https://github.com/qutip/qutip-qip/pull/67#discussion_r664576271)).